### PR TITLE
Add default rendering for unsupported attachments

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/596

### 🎯 Goal
Add default rendering for unsupported attachments.

### 🧪 Manual Testing Notes
1. Enable Location attachments in the Demo App
2. Add a location attachment to a channel
3. Disabled location attachments in the Demo App
4. Go back to the location attachment message
5. It should render as unsupported attachment

### ☑️ Contributor Checklist
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Models/Attachments/AttachmentTypes.swift [hunk 1]
```diff
@@ -303,6 +301,15 @@ public enum AttachmentFileType: String, Codable, Equatable, CaseIterable {
             return false
         }
     }
+
+    public var isUnknown: Bool {
+        switch self {
```
Comment (martinmitrevski) [inline Sources/StreamChat/Models/Attachments/AttachmentTypes.swift:306]: self == .unknown maybe?

File: Sources/StreamChat/Models/ChatMessage.swift [hunk 2]
```diff
@@ -316,7 +316,7 @@ public extension ChatMessage {
 
     /// Returns the attachments of `.file` type.
     ///
-    /// - Important: The `fileAttachments` are loaded lazily and cached to maintain high performance.
+    /// - Important: The `unknownAttachments` are loaded lazily and cached to maintain high performance.
```
Comment (martinmitrevski) [inline Sources/StreamChat/Models/ChatMessage.swift:319]: Hmm, why is this changed, the file attachments are declared here?

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift [hunk 3]
```diff
@@ -270,8 +270,7 @@ open class QuotedChatMessageView: _View, ThemeProvider, SwiftUIRepresentable {
             )
             textView.text = nil
         } else {
-            voiceRecordingAttachmentQuotedPreview.isHidden = true
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:273]: This hiding of views is no longer needed?

File: Sources/StreamChatUI/ChatMessageList/Attachments/Unsupported/UnsupportedAttachmentViewInjector.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        allAttachments.filter {
+            AttachmentType.knownTypes.contains($0.type) == false
+        }
+    }
+}
+
+/// The injector for unknown/unsupported attachments.
+///
+/// By default it renders unsupported attachments as file attachments.
+public class UnsupportedAttachmentViewInjector: AttachmentViewInjector {
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/ChatMessageList/Attachments/Unsupported/UnsupportedAttachmentViewInjector.swift:27]: Can this have an impact on custom attachments from customers? Since we're introducing a new injector, not everyone will start using that.

**Review by martinmitrevski — COMMENTED** [review]
Looks good. Only concern is the impact on custom attachments from customers (which is maybe already handled, but I can't see it straight away).

File: Sources/StreamChatUI/ChatMessageList/Attachments/Unsupported/UnsupportedAttachmentViewInjector.swift (hunk 4 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatMessageList/Attachments/Unsupported/UnsupportedAttachmentViewInjector.swift:27]: No impact whatsover 👍

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift (hunk 3 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:264]: This is now handled by the unsupported view 👍

File: Sources/StreamChat/Models/ChatMessage.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChat/Models/ChatMessage.swift]: Nice catch 👍 Fixed it

File: Sources/StreamChat/Models/Attachments/AttachmentTypes.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChat/Models/Attachments/AttachmentTypes.swift]: Yup makes sense 👍

File: Sources/StreamChatUI/ChatMessageList/Attachments/AttachmentViewCatalog.swift [hunk 5]
```diff
@@ -35,6 +37,10 @@ open class AttachmentViewCatalog {
             return components.isVoiceRecordingEnabled
                 ? components.voiceRecordingAttachmentInjector
                 : components.filesAttachmentInjector
+        } else if attachmentCounts.keys.contains(.unknown) {
+            return components.unsupportedAttachmentInjector
+        } else if attachmentCounts.isEmpty == false {
+            return components.unsupportedAttachmentInjector
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/ChatMessageList/Attachments/AttachmentViewCatalog.swift:43]: why if there are no attachments we return the `unsupportedAttachmentInjector`?

File: Sources/StreamChatUI/ChatMessageList/Attachments/AttachmentViewCatalog.swift [hunk 6]
```diff
@@ -35,6 +37,10 @@ open class AttachmentViewCatalog {
             return components.isVoiceRecordingEnabled
                 ? components.voiceRecordingAttachmentInjector
                 : components.filesAttachmentInjector
+        } else if attachmentCounts.keys.contains(.unknown) {
+            return components.unsupportedAttachmentInjector
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/ChatMessageList/Attachments/AttachmentViewCatalog.swift:41]: what if the attachments contain both known and unknown types?

**Review by martinmitrevski — COMMENTED** [review]
I have 2 questions about the checks in the Attachment Catalog, apart from that looks good.

File: Sources/StreamChatUI/ChatMessageList/Attachments/AttachmentViewCatalog.swift (hunk 5 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatMessageList/Attachments/AttachmentViewCatalog.swift:43]: The condition is the other way around. If there are atttachments, but we don't know the type, then the Unsupported injector is used. (This won't impact customers customizations, since they have their own catalog logic, and if they find their custom type, they user their custom injector). The Demo App is testing all this with the Location attachment

**Review by martinmitrevski — APPROVED** [review]
